src/txHashBuilder.c

```diff
@@ -1303,13 +1300,10 @@ void txHashBuilder_enterCollaterals(tx_hash_builder_t* builder)
 	builder->state = TX_HASH_BUILDER_IN_COLLATERALS;
 }
 
-void txHashBuilder_addCollateral(
-        tx_hash_builder_t* builder,
-        const uint8_t* utxoHashBuffer, size_t utxoHashSize,
-        uint32_t utxoIndex
-)
+void txHashBuilder_addCollateral(tx_hash_builder_t *builder, tx_input_t collInput)
```

Comment:
A pointer here, too.

Reply:
It seems we would want to extract the common code that serializes `tx_input_t` into the tx. (I.e. there would be separate functions for adding inputs and collateral inputs that will handle the state machine logic, but serialization itself can be done by a common static function called from those two.)